src/System.Data.SqlClient/src/System/Data/SqlClient/SqlClientDiagnosticListenerExtensions.cs

```diff
@@ -205,5 +209,72 @@ public static void WriteConnectionCloseError(this DiagnosticListener @this, Guid
                     });
             }
         }
+
+        public static Guid WriteExecuteTransactionBefore(this DiagnosticListener @this, SqlInternalConnection.TransactionRequest transactionRequest, string transactionName,
+                    IsolationLevel iso, SqlInternalTransaction internalTransaction, [CallerMemberName] string operation = "")
```

Comment:
Can you rename `iso` to `isolationLevel` to keep it in sync with the other parameter names being passed?